components/provisioner/internal/model/infrastructure.go

```diff
@@ -58,6 +58,16 @@ func NewAzureInfrastructure(workerCIDR string, azConfig AzureGardenerConfig) *az
 		},
 		Zoned: isZoned,
 	}
+
+	if isZoned && azConfig.input.EnableNatGateway != nil {
+		natGateway := azure.NatGateway{
+			Enabled:                      *azConfig.input.EnableNatGateway,
+			IdleConnectionTimeoutMinutes: util.UnwrapIntOrDefault(azConfig.input.IdleConnectionTimeoutMinutes, 4),
```

Comment:
Maybe we could specify some constant named `defaultNatGatewayIdleConnectionTimeoutMinutes` instead of some mysterious `4`. What do you think?